src/Form/DataField/File.hpp

```diff
@@ -40,6 +40,8 @@ Copyright_License {
 class FileDataField final : public DataField {
   typedef StaticArray<StaticString<32>, 8> PatternList;
 
+  friend class MultiFileDataField;
```

Comment:
Is this really necessary?

Reply:
Yes, the `MultiFileDafield` needs access to `FileDataField`'s private `file` array and to protected member `EnsureLoaded()`.

Reply:
I'm not convinced that this is the best way to solve this. `friend` often means your API is badly designed, and explanations like "but I need to access this private and that protected" do not actually explain anything.

Reply:
This is still here

Reply:
Yes it is indeed still here. The `MultiFileDataField` uses a `FileDataField` internally to keep track of the list of files and other housekeeping tasks. As in, the `MultiFileDataField` simply wraps around the `FileDataField` and keeps track of the selection of multiple files. To let the `FileDataField` keep track of the files, `MultiFileDataField` needs to be able to access the private member array `file`. For most of the `DataField`'s virtual methods it uses the `FileDataField`'s methods. To do that the `MultiFileDataField` needs to be able to call the protected `EnsureLoaded()` method.
A `MultiFileDataField`'s behavior is similar to, but it is not a `FileDataField`. So this way the `FileDataField`'s code is reused without inheriting from it. 
I'm not sure how else to solve this, other than rewriting the `MultFileDataField` as a standalone, with many methods identical to the `FileDataField`'s, which seems wrong to me.

Reply:
An alternative would be to use two `FileDataField`s in the `MultiFileDataField` one for active files and one for the inactive files. This has, I think, the only requirement that a `FileDataField::RemoveFile()` method be added.